sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Percentile.scala

```diff
@@ -138,7 +138,8 @@ case class Percentile(
   override def update(
       buffer: OpenHashMap[Number, Long],
       input: InternalRow): OpenHashMap[Number, Long] = {
-    val key = child.eval(input).asInstanceOf[Number]
+    val scalaValue = CatalystTypeConverters.convertToScala(child.eval(input), child.dataType)
```

Comment:
Should we create a converter and re-use it with `createToScalaConverter(...)` rather than type-dispatching every time?